### ISSUE TITLE
Request sent but user's email address unconfirmed

### ISSUE THREAD

**Issue by RichardTaylor** [mysociety/alaveteli#4674]
The user at
https://www.whatdotheyknow.com/admin/users/137733

has sent a request but their email address is showing as unconfirmed (and so their public user page gives a "Sorry, we couldn't find that page" error). 

It shouldn't be possible to get into this state. 

Do take care not to include personal information on the public GitHub thread.

**Comment by gbp** [comment]
Looking in the production DB I've found 18 users who've managed to send a request without their email being confirmed. So not a huge problem but certainly something we should investigate and try to fix.

The most recent was just a couple of days ago we'll still have the app logs available to us.

**Comment by gbp** [comment]
Here are the logs from the recent request sent by a user with an unconfirmed email:

Going to the new request page for the selected authority: 
`I, [2018-06-20T09:14:19.636498 #31992]  INFO -- : Started GET "/new/[snip]"`

Previewing their request: 
`I, [2018-06-20T09:20:00.842305 #15564]  INFO -- : Started POST "/new"`

Submitting the request: 
`I, [2018-06-20T09:20:13.682273 #31992]  INFO -- : Started POST "/new"`

Request should be stored as PostRedirect but no evidence as the record relating the to token below has already been deleted.

Redirected to sign in/ sign up form:
`I, [2018-06-20T09:20:13.752983 #31992]  INFO -- : Started GET "/profile/sign_in?token=[snip]"`

Submitted Sign up form: 
`I, [2018-06-20T09:20:40.397538 #18690]  INFO -- : Started POST "/profile/sign_up"`

Confirmation email sent: 
`I, [2018-06-20T09:20:40.597719 #18690]  INFO -- : Sent mail to [snip]`

Reloaded the previous page - clicked back in browser history?: 
`I, [2018-06-20T09:22:48.263478 #18690]  INFO -- : Started GET "/profile/sign_in?token=[snip]"`

Sign in: 
`I, [2018-06-20T09:23:01.689664 #31992]  INFO -- : Started POST "/profile/sign_in"`

Post sign in redirect - PostRedirect: 
`I, [2018-06-20T09:23:01.869086 #18690]  INFO -- : Started GET "/new?post_redirect=1"`

Request sent to authority: 
`I, [2018-06-20T09:23:02.049572 #18690]  INFO -- : Sent mail to [snip]`

Viewing new request: 
`I, [2018-06-20T09:23:02.406215 #18690]  INFO -- : Started GET "/request/[snip]"`

**Comment by crowbot** [comment]
What's the first instance of this? Could be useful if we need to track down a regression in the codebase.

**Comment by gbp** [comment]
This is the query I'm using to detect these users:
  `User.joins(:info_requests).where(email_confirmed: false)`.

The first that matches was created on 31 Mar 2012.

**Comment by RichardTaylor** [comment]
I managed to get a request into this state:

A user had 
1. Signed up for the site and been sent a confirmation link, but apparently not received it.
2. Drafted a request, and had been sent a confirmation link, but apparently not received it.

While logged in as an admin I then clicked on the confirmation link for (2) from the user admin page. This sent the request but "Email confirmed" remained false. "Email confirmed" remained false when I clicked the confirmation link from (1) too.  Only selecting the option to log in as the user resulted in "Email confirmed" becoming "True".
